ctms/crud.py

```diff
@@ -270,23 +270,33 @@ def get_contacts_by_any_id(
 
 
 def get_all_acoustic_records_before(
-    db: Session,
-    end_time: datetime,
-    retry_limit: int = 5,
+    db: Session, end_time: datetime, retry_limit: int = 5, batch_limit=None
 ) -> List[PendingAcousticRecord]:
     """
     Get all the pending records before a given date. Allows retry limit to be provided at query time."""
-    pending_records: List[PendingAcousticRecord] = (
+    if batch_limit is None:
+        pending_records: List[PendingAcousticRecord] = (
+            db.query(PendingAcousticRecord)
+            .filter(
+                PendingAcousticRecord.update_timestamp < end_time,
+                PendingAcousticRecord.retry < retry_limit,
+            )
+            .order_by(desc(PendingAcousticRecord.update_timestamp))
+            .all()
```

Comment:
We may want to add a [row lock](https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.with_for_update), like:

```
.with_for_update()
.all()
```

Reply:
I see from PR #234 that we're also using this function to count the size of the backlog, which means we shouldn't unconditionally lock for updates. This suggests it should be another parameter, or even better, split into a crud function to count the backlog size vs get a batch of records.

Reply:
I did not add the row lock, but can still do so; for PR #234 I'll need to go back and rebase on these changes, but I added a separate method to count. _(Still can improve with making a _bg_base_query that can be reused by them both)_

Reply:
let's add the row lock later, we should make sure we understand how to release the lock after the update, rather than leave open forever transactions...